deployments/charts/kuma/templates/cp-deployment.yaml

```diff
@@ -35,64 +35,15 @@ spec:
-              value: "true"
-            - name: KUMA_RUNTIME_KUBERNETES_ADMISSION_SERVER_PORT
-              value: "5443"
-            - name: KUMA_RUNTIME_KUBERNETES_ADMISSION_SERVER_CERT_DIR
-              value: /var/run/secrets/kuma.io/tls-cert
-            - name: KUMA_RUNTIME_KUBERNETES_INJECTOR_CNI_ENABLED
-              value: {{ .Values.cni.enabled | quote }}
-            - name: KUMA_RUNTIME_KUBERNETES_INJECTOR_SIDECAR_CONTAINER_IMAGE
-              value: {{ include "kuma.formatImage" (dict "image" .Values.dataPlane.image "root" $) | quote }}
-            - name: KUMA_INJECTOR_INIT_CONTAINER_IMAGE
-              value: {{ include "kuma.formatImage" (dict "image" .Values.dataPlane.initImage "root" $) | quote }}
-            - name: KUMA_RUNTIME_KUBERNETES_INJECTOR_CA_CERT_FILE
-              value: /var/run/secrets/kuma.io/tls-cert/ca.crt
-            - name: KUMA_DEFAULTS_SKIP_MESH_CREATION
-              value: {{ .Values.controlPlane.defaults.skipMeshCreation | quote }}
-            - name: KUMA_MODE
-              value: {{ .Values.controlPlane.mode | quote }}
-            {{- if .Values.controlPlane.zone }}
-            - name: KUMA_MULTIZONE_REMOTE_ZONE
-              value: {{ .Values.controlPlane.zone | quote }}
-            {{- end }}
-            {{- if .Values.controlPlane.tls.apiServer.secretName }}
-            - name: KUMA_API_SERVER_HTTPS_TLS_CERT_FILE
-              value: /var/run/secrets/kuma.io/api-server-tls-cert/tls.crt
-            - name: KUMA_API_SERVER_HTTPS_TLS_KEY_FILE
-              value: /var/run/secrets/kuma.io/api-server-tls-cert/tls.key
-            {{- end }}
-            {{- if .Values.controlPlane.tls.apiServer.clientCertsSecretName }}
-            - name: KUMA_API_SERVER_AUTH_CLIENT_CERTS_DIR
-              value: /var/run/secrets/kuma.io/api-server-client-certs/
-            {{- end }}
-            {{- if .Values.controlPlane.tls.kdsGlobalServer.secretName }}
-            - name: KUMA_MULTIZONE_GLOBAL_KDS_TLS_CERT_FILE
-              value: /var/run/secrets/kuma.io/kds-server-tls-cert/tls.crt
-            - name: KUMA_MULTIZONE_GLOBAL_KDS_TLS_KEY_FILE
-              value: /var/run/secrets/kuma.io/kds-server-tls-cert/tls.key
-            {{- end }}
-            {{- if .Values.controlPlane.tls.kdsRemoteClient.secretName }}
-            - name: KUMA_MULTIZONE_REMOTE_KDS_ROOT_CA_FILE
-              value: /var/run/secrets/kuma.io/kds-client-tls-cert/ca.crt
-            {{- end }}
-          {{- range $key, $value := .Values.controlPlane.envVars }}
-            - name: {{ $key | upper }}
+          {{-  $defaultEnv := include "kuma.defaultEnv" . | fromYaml | pluck "env" | first }}
+          {{- $defaultEnvDict := dict }}
+          {{- range $index, $item := $defaultEnv }}
+          {{- $name := $item.name | upper }}
+          {{- $defaultEnvDict := set $defaultEnvDict $name $item.value }}
+          {{- end }}
+          {{- $mergedEnv := merge .Values.controlPlane.envVars $defaultEnvDict }}
```

Comment:
We might want to `deepCopy` `.Values.controlPlane.envVars`, as merging into it will mutate it, though this is the only place it's used so more of a nit.

Reply:
So you mean `deepCopy` `.Values.controlPlane.envVars` into a new var and use that for merging?

Reply:
Yeah, exactly, so that the `defaultEnvDict` doesn't modify that variable.